core/src/main/kotlin/org/evomaster/core/problem/rest/service/AbstractRestSampler.kt

```diff
@@ -80,13 +84,43 @@ abstract class AbstractRestSampler : HttpWsSampler<RestIndividual>() {
         val openApiSchema = problem.openApiSchema
 
         if(!openApiURL.isNullOrBlank()) {
-            swagger = OpenApiAccess.getOpenAPIFromURL(openApiURL)
+
+            // first try to retrieve the OpenAPI without authentication
+            swagger = OpenApiAccess.getOpenAPIFromURL(openApiURL, HttpWsNoAuth())
+
+            // if it fails, try to get swagger using each authentication information.
+            // if swagger failed due to unauthorized access, then an empty swagger is retrieved
+            if(swagger.paths == null) {
+
+                var currentAuthenticationIndex = 0
+                val authenticationList:List<AuthenticationDto> = infoDto.infoForAuthentication
+
+                while(currentAuthenticationIndex < authenticationList.size && swagger.paths == null) {
```

Comment:
no need to loop over all auth info. just take the first

Reply:
Hi Andrea,

Thanks for your comment. I have fixed this one.